### ISSUE TITLE
Some XWayland apps don't register keyboard input

### ISSUE THREAD

**Issue by JMoogs** [hyprwm/Hyprland#6253]
### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit 553232a3e4c112c8511309e6b685cb614895e714  ().
Date: 2024-05-25
Tag: , commits:

flags: (if any)


System Information:
System name: Linux
Node name: nixos
Release: 6.9.1
Version: #1-NixOS SMP PREEMPT_DYNAMIC Fri May 17 10:18:09 UTC 2024


GPU information:
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation GA104 [GeForce RTX 3070 Ti] [10de:2482] (rev a1) (prog-if 00 [VGA controller])


os-release: ANSI_COLOR="1;34"
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
BUILD_ID="24.11.20240524.bfb7a88"
DOCUMENTATION_URL="https://nixos.org/learn.html"
HOME_URL="https://nixos.org/"
ID=nixos
IMAGE_ID=""
IMAGE_VERSION=""
LOGO="nix-snowflake"
NAME=NixOS
PRETTY_NAME="NixOS 24.11 (Vicuña)"
SUPPORT_URL="https://nixos.org/community.html"
VERSION="24.11 (Vicuña)"
VERSION_CODENAME="vicuña"
VERSION_ID="24.11"


plugins:
```
</details>

### Bug or Regression?

Bug

### Description

On Black Desert and Town of Salem 2 through proton, keyboard input isn't detected. On other games I tested (Balatro, Lethal Company, Elden Ring, Risk of Rain 2, Minecraft) it seems to work fine. For Black Desert, I can't get past the authentication screen so I can't test properly, but I'd imagine it has something to do with the different type of text input box.

May be related to #6159, though I'm not using gamescope and my mouse seems to work fine.

### How to reproduce

1. Open Hyprland
2. Open Town of Salem 2
3. Create a party from the main menu
4. Try to type in the chat bottom left

or

1. Open Hyprland
2. Turn on 2FA in Black Desert
3. Open Black Desert
4. Attempt to authenticate

### Crash reports, logs, images, videos

The Black Desert authentication input box:
![image](https://github.com/hyprwm/Hyprland/assets/90248874/f1b4abbc-f5f7-4312-a432-7ea71b20c8eb)

The normal behavior is that upon entering a character, the cursor is automatically moved to the next box, ready for you to input the next character.

**Comment by The-Briel-Deal** [comment]
For the authentication input box. Are you able to press anything like Cancel or Confirm? It would be odd to me if mouse worked but keyboard didn't? I'm pretty sure this is an issue of certain windows not grabbing the seat properly (but if mouse works that can't be it I don't think?)

**Comment by nonetrix** [comment]
Gamescope runs XWayland nested inside of Gamescope so it doesn't use Hyprland's inplementation, this seems like entirely separate issue

**Comment by JMoogs** [comment]
> For the authentication input box. Are you able to press anything like Cancel or Confirm?

My mouse works flawlessly in both games as far as I've been able to test.

> Just fyi Gamescope runs XWayland nested inside of Gamescope so it doesn't use Hyprland's inplementation, this seems like entirely separate issue

I thought as much but wasn't sure. Thanks for clarifying.

**Comment by JMoogs** [comment]
Just updated again, seems to be fixed on `cb63398f079b4b4324c04e2e41ba17983d66487c`
